Run forest node in DigitalOcean (#26)
ChainSafe/forest-iac: README.md

- [ ] Exhaustive RPC testing for Forest docker image.
- [ ] Sync testing for PRs.
- [ ] Export testing for PRs.

## Running Forest In DigitalOcean

## Architectural Implementation

> ![Untitled Diagram drawio (8)](https://user-images.githubusercontent.com/47984109/215227510-dac5b8fb-8019-4388-a0e7-d5c432b95d70.png)

Author: Are you referring to the name "Architectural Implementation"?

The flow goes from Terraform for provisioning of the servers and Anisble to run all neccessary installations including forest.

## Requirements 
- RAM: 32GB
- VCPU: 8

> - Startup Disk Size: 100 GB

Reviewer: For startup, we need at least 200G, +100G to download the snapshot and another +100G to fill the DB. After this process, we can delete the snapshot.

- Expected Total Disk Size: > 500 GB

N/B: It's worth to note that some of the naming conventions can be changed to suit your deployment needs. 

To test out the implementation, just access the server with appropiate `ssh` details in this manner `ssh -i ~/id_rsa chainsafe@ip_address`.

Also, out of curiosity, to initiate the infrastructure, run the following:

> - Ensure to add all neccessary enviroment variables for AECCESS_ID and SECRET_KEY into the local machine via the `echo` command. This is needed for the backend and state configurations.

Reviewer: `ACCESS_ID`? Where do you get those variables? How do you add them via `echo` command? Assume the reader is not proficient and definitely assume if you make a typo (s)he will copy-paste it.

Author: On it too.

- Run `make apply` in the terraform directory.
- Run `ansible-playbook forest-docker-run.yaml` in the ansible directory.

## Collaborators

> - [YOUR NAME HERE] - Feel free to contribute to the codebase by resolving any open issues, refactoring, adding new features, writing test cases or any other way to make the project better and helpful to the community. Feel free to fork and send pull requests.

Reviewer: What's the purpose of the `[YOUR NAME HERE]`?

## Questions

> Feel free to reach out to the BlockOps team for more details on how to interact with the infrastructure if the need arises while in deployment.

Reviewer: How to reach the BlockOps team?

Author: I'd change the comment; we should rather go with letting the fellow raise an issue or something

- [ ] Exhaustive RPC testing for Forest docker image.
- [ ] Sync testing for PRs.
- [ ] Export testing for PRs.

## Running Forest In DigitalOcean

## Forest IAC Architectue

![Forest Cloud Infrastructure ](https://user-images.githubusercontent.com/47984109/216006502-eca661d3-2ef8-4c75-aa7a-1740c25abb44.png)

The flow goes from Terraform for provisioning of the servers and Anisble to run all neccessary installations including forest.

## Requirements 
- RAM: 32GB
- VCPU: 8
- Startup Disk Size: 200 GB
- Expected Total Disk Size: > 500 GB

N/B: It's worth to note that most of the naming conventions can be changed to suit your deployment needs especially the names for the resource blocks. In this case; `forest-volume`, `new-key-name`,`spaces-name`,`forest-volume`, `forest`, `forest-firewalls-test` and `hosts`.

To test out the implementation, just access the server with appropiate `ssh` details in this manner `ssh -i ~/.ssh/id_rsa chainsafe@ip_address`. However, if you intend to use another `user` other than `chainsafe`, this can be switched in the Ansible main playbook, just avoid running forest as root.

In order ro implement the infrastructure, run the following:

> - Set Up s3cmd 2.x with DigitalOcean Spaces; you can check [here](https://docs.digitalocean.com/products/spaces/reference/s3cmd/) on more details. This will require `ACCESS_TOKEN` and `SECRET_KEY` but it can be automatically generated from the DigitalOcean console through the Applications & API section.  

Reviewer: ```suggestion
- Set-up `s3cmd` 2.x with DigitalOcean Spaces; you can check [here](https://docs.digitalocean.com/products/spaces/reference/s3cmd/) on more details. This will require `ACCESS_TOKEN` and `SECRET_KEY`; it can be automatically generated from the DigitalOcean console through the Applications & API section.  
```

> - Ensure to add all neccessary enviroment variables by running `echo $SPACES_ACCESS_TOKEN` and `echo $SPACES_SECRET_KEY` while using the values generated in the previous step.

Reviewer: ```suggestion
- add all necessary environment variables by running `echo $SPACES_ACCESS_TOKEN` and `echo $SPACES_SECRET_KEY` while using the values generated in the previous step.
```

> - Ensure to add all neccessary enviroment variables by running `echo $SPACES_ACCESS_TOKEN` and `echo $SPACES_SECRET_KEY` while using the values generated in the previous step.

Reviewer: Printing the variables' values to the console doesn't add them anywhere, no?

Author: Nope it does not

Reviewer: So what do you achieve by printing them?

Author: Oh when you said printing, I thought of something different. Anyways, this stores as part of then `env` values in the local machine while attempting to run the commands.

Reviewer: Sorry, I don't get what you achieve by echoing them. What happens if you don't do it?

Author: If you don't do it then it'll be difficult to setup the backend for the terraform config (which means no spaces and storage of state files)

> - Create a `*.tfvars` file to use the neccessary variable values for terraform running success. Also, be aware that the `keys_name` must be unique.  

Reviewer: What are the necessary values? Let's have a template.
```suggestion
- Create a `*.tfvars` file to use the necessary variable values for terraform running success. Also, be aware that the `keys_name` must be unique.  
```

Author: I think this is not necessary. Anybody that sees the `variables.tf` files can tell what variable values need to go to the `.tfvars` file cos the variables are already declared.
